src/auth/auth-service.ts

```diff
@@ -0,0 +1,58 @@
+/* Facade for keycloak */
+import _ from 'lodash';
+import Keycloak from 'keycloak-js';
+import { getConfig } from '../config';
+import { loadTokens, removeTokens, storeTokens } from './token-store';
+
+let kc: Keycloak.KeycloakInstance;
+
+function setOnAuthSuccess() {
+  const handler = () => storeTokens({ token: kc.token, refreshToken: kc.refreshToken });
+
+  kc.onAuthRefreshSuccess = handler;
+  kc.onAuthSuccess = handler;
+}
+
+function setOnAuthError({ onAuthError }) {
+  if (typeof onAuthError === 'function') {
+    kc.onAuthError = onAuthError;
+    kc.onAuthRefreshError = onAuthError;
+  }
+}
+
+export function initAuth() {
+  kc = Keycloak(getConfig().keycloak);
+
+  kc.onAuthSuccess = () => storeTokens({ token: kc.token, refreshToken: kc.refreshToken });
+  kc.onAuthRefreshSuccess = kc.onAuthSuccess;
+  kc.onAuthError = removeTokens;
+  kc.onAuthRefreshError = removeTokens;
+
+  const { token, refreshToken } = loadTokens();
+  return kc.init({
```

Comment:
Does this return boolean "authenticated" or not? If so, it may be a good idea to check whether authentication is actually successful.

Reply:
promise til undefined. vi returnerer promise, så at vi venter med første render at autorisering er ferdig.

Reply:
vi konfigurerer `onLoad: 'login-required'`, det mener at init skal redirekt til login side hvis ikke er autorisert basert på inisjalisering-token